prereise/gather/winddata/hrrr/hrrr_api.py

```diff
@@ -0,0 +1,115 @@
+        self, start_dt, end_dt, directory, product, selectors=None
+    ):
+        """Iterates from a start datetime (inclusive) to a end datetime (exclusive)
+        at 1 hour steps, downloading data for each intermediary datetime into
+        the directory provided. product and selectors variables are used
+        to control what specific type of data to download. See
+        `this link <https://www.nco.ncep.noaa.gov/pmb/products/hrrr/>`_ to understand
+        more about product and
+        `this link <https://github.com/blaylockbk/Herbie/blob/18945e4c5103386c98d08dcb2de590e2ac14c3d5/docs/user_guide/grib2.rst#how-grib-subsetting-works-in-herbie>`_
+        to understand more about what kind of strings can be passed into selectors
+
+        :param datetime.datetime start_dt: datetime to start at
+        :param datetime.datetime end_dt: datetime to end at
+        :param str directory: file directory to download data into
+        :param str product: info at `this link <https://www.nco.ncep.noaa.gov/pmb/products/hrrr/>`_
+        :param list selectors: list of strings that can be used to narrow down
+            the amount of data downloaded from a specific grib file.
+        """
+        for filename, url in self._filename_url_iter(start_dt, end_dt, product):
+            grib_record_information_list = [GribRecordInfo.full_file()]
+            # first grab index file and figure out which bytes to download
+            if selectors:
+                index_url = generate_index_url(url)
+                response = requests.get(
+                    index_url
+                )  # index files are typically a few kb, so safe to hold in memory
+                raw_record_information_list = response.text.split("\n")
+                index_list = get_indices_that_contain_selector(
+                    raw_record_information_list, selectors
+                )
+                grib_record_information_list = (
+                    GribRecordInfo.generate_grib_record_information_list(
+                        raw_record_information_list, index_list
+                    )
+                )
+
+            with open(directory + filename, "ab") as f:
+                for grib_record_information in grib_record_information_list:
+                    try:
+                        self.downloader.download(
+                            url,
+                            f,
+                            headers={
+                                "Range": f"bytes={grib_record_information.byte_range_header_string()}"
+                            },
+                        )
+                    except Exception:
+                        print(
+                            f"Failed to download data from {url} with byte range {grib_record_information.byte_range_header_string()}"
+                        )
+
+    def download_wind_data(self, start_dt, end_dt, directory):
+        """See prereise.gather.winddata.hrrr.hrrr_api.HrrrApi.download_meteorological_data
```

Comment:
can be referenced as :meth:`prereise.gather.winddata.hrrr.hrrr_api.HrrrApi.download_meteorological_data`